src/Solve.cpp

```diff
@@ -348,6 +348,61 @@ class SolveExpression : public IRMutator2 {
         return expr;
     }
 
+    Expr visit(const Div *op) override {
+        bool old_uses_var = uses_var;
+        uses_var = false;
+        bool old_failed = failed;
+        failed = false;
+        Expr a = mutate(op->a);
+        bool a_uses_var = uses_var;
+        bool a_failed = failed;
+
+        internal_assert(!is_const(op->a) || !a_uses_var)
+                << op->a << ", " << uses_var << "\n";
+
+        uses_var = false;
+        failed = false;
+        Expr b = mutate(op->b);
+        bool b_uses_var = uses_var;
+        bool b_failed = failed;
+        uses_var = old_uses_var || a_uses_var || b_uses_var;
+        failed = old_failed || a_failed || b_failed;
+
+        const Add *add_a = a.as<Add>();
+        const Sub *sub_a = a.as<Sub>();
+        const Mul *mul_a = a.as<Mul>();
+
+        Expr expr;
+        if (a_uses_var && !b_uses_var) {
+            if (add_a && !a_failed &&
+                can_prove(add_a->a / b * b == add_a->a)) {
```

Comment:
Interesting. The correctness_likely test will time out inside this `can_prove` call. I am not sure how to fix. BTW, on the new simplifier branch, this code work (Woo, good).